src/artifacts.js

```diff
@@ -509,6 +509,16 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
       key: artifact.details.key,
     };
 
+    res.set('x-amz-meta-runid', runId);
+    res.set('x-amz-meta-taskid', taskId);
+    res.set('x-amz-meta-name', name);
```

Comment:
I see where these are set... I don't think they are useful, and I don't think that client should use them any way..

They might nice to have if scanning the S3 bucket.

Reply:
They are for the purposes of scanning the bucket and also in case we start getting artifacts and can't figure out where they're from.  Having these as headers makes that simple, rather than having to possibly hash the transfer-sh256 and scan the whole artifacts table looking for an artifact that has the transfer-256 that matches, we can simply look at the header.  Since the creation of these headers are signed by remotely-signed-s3, we can trust them to be correct.

Reply:
That doesn't explain why they are part of the redirect request.


Having them in S3 does no harm. Here it just seems weird to have them. I particularly don't like having things like `x-amz-...` returned from our APIs.

Reply:
it's just easier to have the same name for the comparisons if we use the same name.